x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/cert/CertificateInfo.java

```diff
@@ -64,6 +67,11 @@ public CertificateInfo(StreamInput in) throws IOException {
         this.serialNumber = in.readString();
         this.hasPrivateKey = in.readBoolean();
         this.expiry = Instant.ofEpochMilli(in.readLong()).atZone(ZoneOffset.UTC);
+        if (in.getVersion().onOrAfter(Version.V_8_4_0)) {
+            this.issuer = in.readString();
+        } else {
+            this.issuer = "[unknown]";
```

Comment:
This object does not get sent across the wire. So this is indeed just nitpicking. That said, I'd suggest setting the issuer to `null` in this case and skip it in `toXContent` method when it is `null` to (1) avoid having an arbitrary default value; (2) closer to the existing behaviour.

Reply:
First of all, thanks for indulging me with this discussion. We all agree that it does not really matter for this PR (it is nitpicking). Therefore I am not trying to make a strong case here and also do not want to block it from merging as is. 

> If we threw an exception here, for 8.7+ you could no longer use that existing API in an 8.4-,8.7+ mixed clusters. That would likely be undesirable and possibly a bug, so in 8.7 we would need add support for mixed clusters for these version ranges

That was my intention. My thinking is that we address it when it is actually needed. Ideally a test case should catch it in 8.7 before releasing. So it won't become a production bug. But there is a good chance that the test coverage is not sufficient and it gets slipped into a released version. So I admit it is a bit risky.

> It can be argued that we could remove the code/test for transport serialization here

I did not mean that. The proposed exception is only thrown if serialisation is performed cross specific versions. Please correct me if I am wrong, I think most serialisation tests do not test for cross versions? So no need to change the serialisation tests.

> Keeping the "" as default from elder versions, in future cases, you may actually get back an "" as the value

I do not particularly like using empty string as the special value. I feel it has pros and cons compared to `null`. I don't think `null` is great. But as a special value, `null` is at the language level (and fails loudly) while the empty string is our choosing. 

Again, I have no strong views, just clarifying my thoughts. Hopefully it could be more relevant in some future context. Thanks!

Reply:
tl;dr - will update PR to conditionally omit the empty string return value

@ywangd and I met in person yesterday and here are the notes and outcome. 

We agreed:
* it doesn't really matter for this PR but this conversation helps with future alignment where it could matter
* (we agreed) that the purpose of the transport serialization while not in use toady is added to classes like these to help enable future cases 
* (we agreed) this pattern will likely repeat itself since much of the security code runs on the coordinating node (i.e. no transport serialization) 

We ruled out:
* throwing an exception since it runs counter to the goal to enable future use cases. Additionally our mixed cluster testing for x-pack testing is also [lacking](https://github.com/elastic/elasticsearch/issues/31096). However in general, throwing exceptions when the decision is unclear and can be deferred is generally a good approach, but we don't want to defer here. 

So this leave two areas of discussion:
* external serialization
* internal representation

For external serialization we can emit a key/value pairing where the value is null, "", or something like "unknown". In this context, the object we are serializing represents a certificate and the field is the issuer. It is not possible for a certificate to not have issuer, so strictly speaking emitting anything but the acutal value for the issuer is incorrect. So we believe correct answer for how to represent this externally is to omit the value if can not determine the value. It adds a tiny amount of additional code that is may never be used. This is OK because but it is easily testable and is in line with the goals of ensuring that these POJO type classes are ready for future expanded use cases. I will update the PR with this change.

For internal representation, it does not really matter. It probably makes the most sense to represent the variable as null since it is missing, but that forces a mismatch of the constructors. Where one constructor sets the variable to be null and the other enforces it as non-null. Since the non-null expectations are the primary and current workflow we will optimize for that an internally represent as "". 

I know this is way overkill for this PR, but getting pedantic now on an issue that does not matter means that we don't need to distract from future more meaningful PRs. I will look to updating some internal documentation with this kind of guidance. 

Thanks again @yangd and @gwbrown for the conversation here, and @sethmlarson for updating our internal REST API guidance for how to handle null/empty values.